docs/operational_guide/bootstrapping.md

```diff
@@ -0,0 +1,112 @@
+
+```
+    ┌─────────────────┐          ┌─────────────────┐        ┌─────────────────┐
+    │     Host 1      │          │     Host 2      │        │     Host 3      │
+────┴─────────────────┴──────────┴─────────────────┴────────┴─────────────────┴───
+┌─────────────────────────┐   ┌───────────────────────┐   ┌──────────────────────┐
+│                         │   │                       │   │                      │
+│                         │   │                       │   │                      │
+│  Shard 1: Initializing  │   │ Shard 1: Available    │   │  Shard 1: Available  │
+│  Shard 2: Initializing  │   │ Shard 2: Available    │   │  Shard 2: Available  │
+│  Shard 3: Initializing  │   │ Shard 3: Available    │   │  Shard 3: Available  │
+│                         │   │                       │   │                      │
+│                         │   │                       │   │                      │
+└─────────────────────────┘   └───────────────────────┘   └──────────────────────┘
+```
+
+Note that a bootstrap consistency level of majority is the default value, but can be modified by changing the value of the key "m3db.client.bootstrap-consistency-level" in [etcd](https://coreos.com/etcd/) to one of: "none", "one", "unstrict_majority" (attempt to read from majority, but settle for less if any errors occur), "majority" (strict majority), and "all".
+
+**Note**: Any bootstrappers configuration that does not include the peers bootstrapper will be unable to handle dynamic placement changes of any kind.
+
+### Uninitialized Topology Bootstrapper
+
+The purpose of the uninitialized topology bootstrapper is to succeed bootstraps for all time ranges for shards that have never been completely bootstrapped (at a cluster level). This allows us to run the default bootstrapper configuration of: `filesystem,commitlog,peers,topology_uninitialized` such that the `filesystem` and `commitlog` bootstrappers are used by default in node restarts, the peer bootstrapper is used for node adds/removes/replaces, and bootstraps still succeed for brand new placement where both the `commitlog` and `peers` bootstrappers will be unable to succeed any bootstraps. In other words, the uninitialized topology bootstrapper allows us to place the `commitlog` bootstrapper *before* the `peers` bootstrapper and still succeed bootstraps with brand new placements without resorting to using the noop-all bootstrapper which suceeds bootstraps for all shard/time-ranges regardless of the status of the placement.
```

Comment:
maybe just say `uninitialized_topology` instead of "uninitialized topology" in this para?

Reply:
same thing, say `peers` bootstrapper, not "peer bootstrapper"

Reply:
fixed